lib/commands/recordscreen.js

```diff
@@ -0,0 +1,129 @@
+import _ from 'lodash';
+import { retryInterval } from 'asyncbox';
+
+let commands = {}, extensions = {};
+
+const RETRY_PAUSE = 1000;
+
+/**
+* record the display of devices running Android 4.4 (API level 19) and higher.
+* It records screen activity to an MPEG-4 file. Audio is not recorded with the video file.
+*
+* @param filePath the video file name
+*                 for example, "/sdcard/demo.mp4"
+* @param videoSize the format is widthxheight.
+*                  if it is "default", the default value is the device's native display resolution (if supported),
+*                  1280x720 if not. For best results,
+*                  use a size supported by your device's Advanced Video Coding (AVC) encoder.
+*                  for example, "1280x720"
+* @param timeLimit the maximum recording time, in seconds. if it is -1, the default and maximum value is 180 (3 minutes).
+* @param bitRate the video bit rate for the video, in megabits per second.
+*                if it is -1, the default value is 4Mbps. You can increase the bit rate to improve video quality,
+*                but doing so results in larger movie files.
+*                for example, 6000000
+*
+*/
+commands.startRecordingScreen = async function ( filePath, videoSize, timeLimit, bitRate ) {
+  let cmd, data, length, indexNoSuchFile;
+
+  data = await this.adb.getApiLevel();
+
+  //this function is suppported on the device running android 4.4(api level 19)
+  if (data < 19) throw new Error('The screenrecord command is able to record the display of devices running Android 4.4 (API level 19) and higher.');
+
+  cmd = ['ls', `${filePath}`, '|', 'grep', 'No such file'];
+  data = await this.adb.shell(cmd); 
+
+  //if there's same file in the path, then thorws error
+  if ( _.size(data) === 0) throw new Error(`${filePath} is already exist.`);
+
+
+  //make adb command 
+  cmd = ['screenrecord', `${filePath}`];
+
+  if ( videoSize !== "default" ) {
+    cmd.push('--size', videoSize);
+  }  
+
+  if ( timeLimit !== -1 ) {
+    cmd.push('--time-limit', timeLimit);
+  } 
+
+  if ( bitRate !== -1 ) {
+    cmd.push('--bit-rate', bitRate);
+  } 
+
+  //call the command asynchronously 
+  data = this.adb.shell(cmd); 
+
+  
+  //there is the delay time to start recording the screen, so, wait until it is ready.
+  //the ready condition is
+  //1. check the movie file is created
+  //2. check it is started to capture the screen 
+  return await retryInterval(RETRY_PAUSE, RETRY_PAUSE, async () => {
+    cmd = ['ls', '-al', `${filePath}`]; 
+    data = await this.adb.shell(cmd);
+    indexNoSuchFile = data.indexOf("No such file");
```

Comment:
If the file is not found, throw an exception. This is what triggers the `retryInterval` to pause and then go into the next iteration. Otherwise this will only happen once.